docs/contributing/sleep_deep_sleep.md

```diff
@@ -1,3 +1,28 @@
-### Sleep and deep sleep
+# Sleep HAL API
 
-[A document explaining sleep and deep sleep]
+mbed OS defines two sleep modes for HAL:
+
+- Sleep
+- Deep sleep
+
+Each target should document in their implementation:
+
+- The target's mode description for each mode (how target's mode map to our sleep modes)
+- Wake-up latency for each mode
+- Wake-up sources for each mode
+
+## Sleep
+
+The core system clock is disabled, both the low and high precision clocks are enabled, RAM is retained.
+
+1. Wake-up sources - any interrupt source should be able to wake-up the MCU
+1. Latency - should wake-up within 10 us
+
+## Deep sleep
+
+The core system clock is disabled. The low precision clocks are enabled, RAM is retained.
+
+1. Wake-up sources - RTC, low power ticker or GPIO should be able to wake-up the MCU
+1. Latency - should wake-up within 5 ms
+
+The deep sleep latency (5 ms) was chosen as the higher limit of the vendors we support. Most of targets have wake-up latency for deep sleep within few microseconds. This wake-up time does not include reinitializing clocks or other related configuration to return to previous state after the deep sleep.
```

Comment:
I'm not sure I like that. We should either give a strict limit to go back to previous state or explicitly mark it as undefined.

Reply:
I believe our intention was to have those 10ms (we increased it from 5) as a strict limit. Most of MCU can wake up within few microseconds, the part that takes are clocks related config reinit, up to few milliseconds. Shall we update this accordingly ?